src/App.js

```diff
@@ -14,11 +14,74 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.Component {
+  state = {
+    selectedGoods: [],
+  }
+
+  selectBtnClick = (arr, good) => {
```

Comment:
```suggestion
  selectBtnClick = (good) => {
```
Ще один масив тут не потрібен. Просто використовуй setState для запису нового елементу в стейт.